File: backend/src/main/java/fr/pedalons/dto/admin/CreateGpsCredentialRequest.java
package fr.pedalons.dto.admin;

import fr.pedalons.dto.validation.ValidateSchema;
import fr.pedalons.enums.GpsServiceType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.jspecify.annotations.Nullable;

@Schema(description = "Request to create a new GPS credential")
@ValidateSchema
public record CreateGpsCredentialRequest(
    @NotNull @Schema(description = "GPS service type", required = true) GpsServiceType serviceType,
    @NotBlank @Size(max = 255) @Schema(description = "OAuth client ID", required = true)
        String clientId,
    @Nullable
        @Size(max = 500)
        @Schema(description = "OAuth client secret (nullable for PKCE services)")
        String clientSecret,
    @Schema(description = "Whether credential is active") boolean active) {}

File: frontend/src/api/dto/createGpsCredentialRequest.ts
import type { GpsServiceType } from './gpsServiceType.ts'

/**
 * Request to create a new GPS credential
 */
export interface CreateGpsCredentialRequest {
  /** GPS service type */
  serviceType: GpsServiceType
  /**
   * OAuth client ID
   * @maxLength 255
   * @pattern \S
   */
  clientId: string
  /**
   * OAuth client secret (nullable for PKCE services)
   * @maxLength 500
   */
  clientSecret?: string
  /** Whether credential is active */
  active?: boolean
}
